src/main.js

```diff
@@ -236,9 +233,26 @@ class RoundSlider extends LitElement {
     const rect = this.shadowRoot.querySelector("svg").getBoundingClientRect();
     const scale = Math.max(rect.width, rect.height);
     this._scale = 2/scale;
-  }
-
 
+    if(changedProperties.has("readonly") || changedProperties.has("disabled")) {
+      if(this._showHandle && !this.disabled && !this.attachedListeners) {
+        document.addEventListener('mouseup', this.dragEnd.bind(this));
+        document.addEventListener('touchend', this.dragEnd.bind(this), {passive: false});
+        document.addEventListener('mousemove', this.drag.bind(this));
+        document.addEventListener('touchmove', this.drag.bind(this), {passive: false});
+        document.addEventListener('keydown', this._keyStep.bind(this));
+        this.attachedListeners = true;
+      }
+      else if((!this._showHandle || this.disabled) && this.attachedListeners) {
+        document.removeEventListener('mouseup', this.dragEnd.bind(this));
+        document.removeEventListener('touchend', this.dragEnd.bind(this), {passive: false});
+        document.removeEventListener('mousemove', this.drag.bind(this));
+        document.removeEventListener('touchmove', this.drag.bind(this), {passive: false});
+        document.removeEventListener('keydown', this._keyStep.bind(this));
+        this.attachedListeners = false;
+      }
+    }
+  }
```

Comment:
Why not just return early from the listeners if they're not needed?
Am I missing something?
Will it affect how events pass through the element?

Reply:
Hmm we can definitely do that but I would think its better for performance if we don't need to be listening to these events then we shouldn't be

Reply:
I'll just trust you on that one.

Reply:
I mean I am no expert :) We can ask the expert Bram/Paulus. I would be curious if it actually is better to remove them. Or if it is the same

Reply:
Doesn't matter, because this doesn't even work... 😝